app/tests/helpers/mock_models.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+import tweepy
+
+
+class MockData:
+    user_file = open("app/tests/data/user.json")
+    data = json.load(user_file)
+    user = None
+    tweets = None
+    user_file.close()
+
+    @classmethod
+    def user_info(cls):
+        if cls.user is None:
+            cls.user = tweepy.User(
+                data={x: cls.data[x] for x in cls.data if x not in ["tweets"]}
+            )
+        return cls.user
+
+    @classmethod
+    def tweets_info(cls):
+        if cls.tweets is None:
+            cls.tweets = [tweepy.Tweet(data=x) for x in cls.data["tweets"]]
+        return cls.tweets
+
+
+class MockResponse:
+    def __init__(self, status_code=None, json_data={}, reason="None"):
+        self.status_code = status_code
+        self.json_data = json_data
+        self.reason = reason
+
+    def json(self):
+        return self.json_data
+
+
+class MockPaginator:
+    """
+    This paginator returns:
+    - Tweets if get all tweets (for TwitterScraper.get_tweet_info)
+    - Nothing if get recent tweets (for TwitterScraper.get_frequency)
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.flatten_value = [] if "start_time" in kwargs else MockData.tweets_info()
+
+    def flatten(self, *args, **kwargs):
+        return self.flatten_value
```

Comment:
Or you could mock `tweepy.Client` method to return pagination token you want

Reply:
Im completely clueless

Reply:
I mean you don't have to mock `tweepy.Paginator`, you could mock method of `tweepy.Client` that we use directly. For example, for function call `tweepy.Paginator(api_v2.get_users_tweets, ...)`, `tweepy.Paginator` just call method `get_users_tweets` of `tweepy.Client` to get response, so we could mock `get_users_tweets` method directly to just return the tweets we want.

Reply:
so you dived into their code :v? Great idea actually just i'm pretty happy with the current mocking situation (and lazy). Maybe refac if there's future hickups.

Also the fact that if we mock something so deep inside the outside package (tweepy in this case), it's hard to know wtf is going on when you read the mock code. So I think the better choice is to mock something that actually can be seen used in `OUR` code instead (`paginator` instead of `cliend`).